src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIResponse.java

```diff
@@ -114,14 +114,18 @@ public boolean isSuccess() {
     return isSuccess;
   }
 
-  private void checkRetryable() {
-    Gson gson = new Gson();
-    JsonObject jo = gson.fromJson(this.responseBody, JsonObject.class);
-    if (jo.get(ServiceNowConstants.STATUS) != null &&
-      jo.get(ServiceNowConstants.STATUS).getAsString().equals(ServiceNowConstants.FAILURE) &&
-      jo.getAsJsonObject(ServiceNowConstants.ERROR).get(ServiceNowConstants.MESSAGE).getAsString()
-        .contains(ServiceNowConstants.MAXIMUM_EXECUTION_TIME_EXCEEDED)) {
-      isRetryable = true;
+  private void checkRetryable() throws IOException {
+    try {
+      Gson gson = new Gson();
+      JsonObject jo = gson.fromJson(this.responseBody, JsonObject.class);
+      if (jo.get(ServiceNowConstants.STATUS) != null &&
+        jo.get(ServiceNowConstants.STATUS).getAsString().equals(ServiceNowConstants.FAILURE) &&
+        jo.getAsJsonObject(ServiceNowConstants.ERROR).get(ServiceNowConstants.MESSAGE).getAsString()
+          .contains(ServiceNowConstants.MAXIMUM_EXECUTION_TIME_EXCEEDED)) {
+        isRetryable = true;
```

Comment:
This method is only called from the constructor. Instead of setting a mutable field, it is better to have this be a static method and return a boolean indicating if it is retryable or not such that the constructor can set it to a final field.

Also, all other fields of this class should be final too.